File: Src/Sst/LuaTemplate/Lib/Module.lua
local Classes = require "Classes"

local Module = Classes.class()

function Module:init(child)
	self.IModule = LuaProxy:CreateModule(child,LuaHelper)

	self.ModuleName = ""
	self.ServiceId = ""
end

-- abstract
function Module:Initialize(server,cacheManager,controllerComponentManager)
	-- Body
end

-- abstract
function Module:InitFinish(server,cacheManager,controllerComponentManager)
	-- Body
end

-- abstract
function Module:Dispose(cacheManager,controllerComponentManager)
	-- Body
end

-- abstract
function Module:Dispose()
	-- Body
end

-- abstract
function Module:Finish(server,cacheManager,controllerComponentManager)
	-- Body
end

-- abstract
function Module:Accepted(peer,readStream,writeStream)
	-- Body
	return true
end

-- abstract
function Module:Connected(peer,readStream)
	-- Body
end

-- abstract
function Module:Disconnected(peer)
	-- Body
end

function Module:AddController(func)
	self.IModule:AddController(func)
end

function Module:AddPacket(interface,func)
	self.IModule:AddPacket(interface,func)
end

return Module

File: Src/Sst/PythonTemplate/Lib/Module.py

import clr
import sys
clr.AddReference("Sss")

from Sss.SssScripts.Python import PythonProxy

class Module:
	def __init__(self):
		self.IModule = PythonProxy.CreateModule(self,sys.PythonHelper)
		self.ServiceId = ""

	def Initialize(self,server,cacheManager,controllerComponentManager):
		pass
	
	def InitFinish(self,server,cacheManager,controllerComponentManager):
		pass

	def Finish(self,server,cacheManager,controllerComponentManager):
		pass

	def Dispose(self,cacheManager,controllerComponentManager):
		pass

	def Accepted(self,peer,readStream,writeStream):
		return True

	def Connected(self,peer,readStream):
		pass

	def Disconnected(self,peer):
		pass

	def AddController(self,generator):
		return self.IModule.AddController(generator)

	def AddPacket(self,interface,implement):
		self.IModule.AddPacket(interface,implement)